### PR TITLE
Add sara r4 compile warning check

### PR DESCRIPTION
**Pull request by chinglee-iot**
* Fix SARA R4 cellular module compile warning to comply with C90

### CODE REVIEW

File: modules/sara_r4/cellular_r4_api.c [hunk 1]
```diff
@@ -589,7 +559,7 @@ CellularError_t Cellular_SocketRecv( CellularHandle_t cellularHandle,
     }
     else
     {
-        sessionId = ( uint32_t ) socketHandle->pModemData;
+        sessionId = _Cellular_GetSessionId( pContext, socketHandle->socketId );
```
Comment (andysun2015) [inline modules/sara_r4/cellular_r4_api.c:562]: The previous implementation never works or due to the changes in FreeRTOS-Cellular-Interface?

File: test/unit-test/CMakeLists.txt [hunk 2]
```diff
@@ -205,3 +205,42 @@ create_test(${utest_name}
...
+        )
+
+set(utest_dep_list "")
+list(APPEND utest_dep_list
+            ${cellular_common_real_name}
+            ${cellular_sara_r4_real_name}
+        )
+
+set(utest_name "${project_name}_sara_r4_utest")
+set(utest_source "${project_name}_sara_r4_utest.c")
```
Comment (andysun2015) [inline test/unit-test/CMakeLists.txt:240]: I saw there is an empty file `cellular_sara_r4_utest.c` to do warning check, I think your purpose is to build the module code, is it necessary to add the empty file for this purpose?

Comment (chinglee-iot) [reply test/unit-test/CMakeLists.txt]: Move the module compilation CI check to separate PR.

File: modules/sara_r4/cellular_r4_api.c (hunk 1 above)
Comment (chinglee-iot) [reply modules/sara_r4/cellular_r4_api.c:562]: The previous implementation is not comply with C90. The function _Cellular_GetSessionId is defined in module implementation. It should be changed in one PR for all the function name in this module.

File: modules/sara_r4/cellular_r4.c [hunk 3]
```diff
@@ -490,11 +492,58 @@ uint32_t _Cellular_GetSocketId( CellularContext_t * pContext,
         cellularStatus = CELLULAR_BAD_PARAMETER;
     }
 
-    if( ( cellularStatus == CELLULAR_SUCCESS ) &&
-        ( sessionId >= MIN_TCP_SESSION_ID ) && ( sessionId <= MAX_TCP_SESSION_ID ) )
+    if( ( cellularStatus == CELLULAR_SUCCESS ) && ( sessionId <= MAX_TCP_SESSION_ID ) )
```
Comment (andysun2015) [inline modules/sara_r4/cellular_r4.c:495]: The condition ( sessionId >= MIN_TCP_SESSION_ID ) can't be ignored.

Comment (chinglee-iot) [reply modules/sara_r4/cellular_r4.c:495]:
```
#define MIN_TCP_SESSION_ID          ( 0U )
uint32_t _Cellular_GetSocketId( CellularContext_t * pContext,
                                uint8_t sessionId )
```
uint8_t is always >= 0

File: modules/sara_r4/cellular_r4.c [hunk 4]
```diff
@@ -490,11 +492,58 @@ uint32_t _Cellular_GetSocketId( CellularContext_t * pContext,
...
     {
         socketIndex = pModuleContext->pSessionMap[ sessionId ];
     }
 
     return socketIndex;
 }
+
+/*-----------------------------------------------------------*/
+
+uint32_t _Cellular_GetSessionId( CellularContext_t * pContext,
```
Comment (andysun2015) [inline modules/sara_r4/cellular_r4.c:505]: As discussed, this API should follow the [coding convention from embedded c-sdk](https://docs.aws.amazon.com/embedded-csdk/latest/lib-ref/docs/doxygen/output/html/guide_developer_styleguide.html) since it exist in FreeRTOS repo, but it can be put another PR.

File: modules/sara_r4/cellular_r4.c [hunk 5]
```diff
@@ -490,11 +492,58 @@ uint32_t _Cellular_GetSocketId( CellularContext_t * pContext,
...
+        cellularStatus = CELLULAR_BAD_PARAMETER;
+    }
+    else
+    {
+        cellularStatus = _Cellular_GetModuleContext( pContext, ( void ** ) &pModuleContext );
+    }
+
+    if( cellularStatus == CELLULAR_SUCCESS )
+    {
+        for( sessionId = 0; sessionId < TCP_SESSION_TABLE_LEGNTH; sessionId++ )
```
Comment (andysun2015) [inline modules/sara_r4/cellular_r4.c:529]:
```suggestion
        for( sessionId = 0U; sessionId < TCP_SESSION_TABLE_LEGNTH; sessionId++ )
```

File: modules/sara_r4/cellular_r4.c (hunk 3 above)
Comment (andysun2015) [reply modules/sara_r4/cellular_r4.c:495]: But now I see #define MIN_TCP_SESSION_ID          ( 0 ) instead of #define MIN_TCP_SESSION_ID          ( 0U )
One of them need to be corrected.

File: modules/sara_r4/cellular_r4_api.c [hunk 6]
```diff
@@ -1039,7 +1036,7 @@ CellularError_t Cellular_DeactivatePdn( CellularHandle_t cellularHandle,
         {
             LogDebug( ( "Cellular_DeactivatePdn: Listing operator and context details below." ) );
 
-            for( int i = 0; i < MAX_PDP_CONTEXTS - 1; i++ )
+            for( i = 0; i < ( MAX_PDP_CONTEXTS - 1 ); i++ )
```
Comment (andysun2015) [inline modules/sara_r4/cellular_r4_api.c:1039]:
```suggestion
            for( i = 0U; i < ( MAX_PDP_CONTEXTS - 1U ); i++ )
```

File: modules/sara_r4/cellular_r4_api.c [hunk 7]
```diff
@@ -2583,13 +2609,14 @@ CellularError_t Cellular_SetPdnConfig( CellularHandle_t cellularHandle,
         {
             LogDebug( ( "Cellular_SetPdnConfig: Listing operator and context details below." ) );
 
-            for( int i = 0; i < MAX_PDP_CONTEXTS - 1; i++ )
+            for( i = 0; i < ( MAX_PDP_CONTEXTS - 1 ); i++ )
```
Comment (andysun2015) [inline modules/sara_r4/cellular_r4_api.c:2612]:
```suggestion
            for( i = 0U; i < ( MAX_PDP_CONTEXTS - 1U ); i++ )
```

File: test/unit-test/CMakeLists.txt (hunk 2 above)
Comment (andysun2015) [reply test/unit-test/CMakeLists.txt]: Sure, that's what we discuss before.

File: modules/sara_r4/cellular_r4.c (hunk 4 above)
Comment (chinglee-iot) [reply modules/sara_r4/cellular_r4.c:505]: Update all in another PR.

File: modules/sara_r4/cellular_r4.c (hunk 3 above)
Comment (chinglee-iot) [reply modules/sara_r4/cellular_r4.c:495]: Sorry for typo it is 
```
#define MIN_TCP_SESSION_ID          ( 0 )
```
uint8_t is always  >= 0. What is the point to compare uint8_t >= 0?

```
./cellular_r4_api.c:451:                if( ( tempValue >= MIN_TCP_SESSION_ID ) && ( tempValue <= MAX_TCP_SESSION_ID ) )
./cellular_r4_urc_handler.c:395:                if( ( tempValue >= MIN_TCP_SESSION_ID ) && ( tempValue <= MAX_TCP_SESSION_ID ) )
./cellular_r4_urc_handler.c:490:                if( ( tempValue >= MIN_TCP_SESSION_ID ) && ( tempValue <= MAX_TCP_SESSION_ID ) )
./cellular_r4_urc_handler.c:554:                if( ( tempValue >= MIN_TCP_SESSION_ID ) && ( tempValue <= MAX_TCP_SESSION_ID ) )
```
The usage here is to compare int32_t ( result of Cellular_ATStrtoi ).

Comment (chinglee-iot) [reply modules/sara_r4/cellular_r4.c:495]: Original warning message.
```
if( ( cellularStatus == CELLULAR_SUCCESS ) &&
        ( sessionId >= MIN_TCP_SESSION_ID ) && ( sessionId <= MAX_TCP_SESSION_ID ) )
```

Comment (chinglee-iot) [reply modules/sara_r4/cellular_r4.c]: Update the type comparison MAX_TCP_SESSION_ID in next commit.

**Review by andysun2015 — APPROVED** [review]
Looks good to me.

**Review by n9wxu — APPROVED** [review]
